pandas/core/internals/blocks.py

```diff
@@ -625,13 +651,17 @@ def downcast(self, dtypes=None) -> List[Block]:
         elif dtypes != "infer":
             raise AssertionError("dtypes as dict is not supported yet")
 
-        # operate column-by-column
-        # this is expensive as it splits the blocks items-by-item
-        def f(mask, val, idx):
-            val = maybe_downcast_to_dtype(val, dtype="infer")
-            return val
+        return self._downcast_2d()
 
-        return self.split_and_operate(None, f, False)
+    @maybe_split
```

Comment:
I find using a decorator & having _split very confusing.

Reply:
hah i find split_and_operate (the non-decorator alternative) confusing.  maybe it can be simplified now that putmask no longer uses it (so one or more of its args can be removed)

Reply:
so i can re-write split_and_operate to be basically just the `if self.ndim == 2 and self.shape[0] != 1` part of `maybe_split`, but then every method that is decorated in this PR looks like:

```
def foo(self, *args, **kwargs):
    if self.ndim == 2 and self.shape[0] != 1:
       return self._split_and_operate(Block.foo, *args, **kwargs)
    [...]
```

and avoiding that repetition seems like exactly the use case for a decorator

Reply:
tried this out and it gets more verbose bc we have to write out *args/**kwargs each time

Reply:
mixing these approaches just makes this more confusing. pick.

Reply:
ok, will update shortly.